ctapipe_io_lst/calibration.py

```diff
@@ -0,0 +1,778 @@
+                r1.selected_gain_channel = None
+
+            # store calibration data needed for dl1 calibration in ctapipe
+            # first drs4 time shift (zeros if no calib file was given)
+            time_shift = self.get_drs4_time_correction(
+                tel_id, self.first_cap[tel_id],
+                selected_gain_channel=selected_gain_channel,
+            )
+
+            # time shift from flat fielding
+            if self.mon_data is not None and self.add_calibration_timeshift:
+                time_corr = self.mon_data.tel[tel_id].calibration.time_correction
+                # time_shift is subtracted in ctapipe,
+                # but time_correction should be added
+                if selected_gain_channel is not None:
+                    time_shift -= time_corr[r1.selected_gain_channel, pixel_index].to_value(u.ns)
+                else:
+                    time_shift -= time_corr.to_value(u.ns)
+
+            event.calibration.tel[tel_id].dl1.time_shift = time_shift
+
+            # needed for charge scaling in ctpaipe dl1 calib
+            if selected_gain_channel is not None:
+                relative_factor = np.empty(n_pixels)
+                relative_factor[r1.selected_gain_channel == HIGH_GAIN] = self.calib_scale_high_gain.tel[tel_id]
+                relative_factor[r1.selected_gain_channel == LOW_GAIN] = self.calib_scale_low_gain.tel[tel_id]
+            else:
+                relative_factor = np.empty((n_gains, n_pixels))
+                relative_factor[HIGH_GAIN] = self.calib_scale_high_gain.tel[tel_id]
+                relative_factor[LOW_GAIN] = self.calib_scale_low_gain.tel[tel_id]
+
+            event.calibration.tel[tel_id].dl1.relative_factor = relative_factor
+
+    @staticmethod
+    def _read_calibration_file(path):
+        """
+        Read the correction from hdf5 calibration file
+        """
+        mon = MonitoringContainer()
+
+        with tables.open_file(path) as f:
+            tel_ids = [
+                int(key[4:]) for key in f.root._v_children.keys()
+                if key.startswith('tel_')
+            ]
+
+        for tel_id in tel_ids:
+            with HDF5TableReader(path) as h5_table:
+                base = f'/tel_{tel_id}'
+                # read the calibration data
+                table = base + '/calibration'
+                next(h5_table.read(table, mon.tel[tel_id].calibration))
+
+                # read pedestal data
+                table = base + '/pedestal'
+                next(h5_table.read(table, mon.tel[tel_id].pedestal))
+
+                # read flat-field data
+                table = base + '/flatfield'
+                next(h5_table.read(table, mon.tel[tel_id].flatfield))
+
+                # read the pixel_status container
+                table = base + '/pixel_status'
+                next(h5_table.read(table, mon.tel[tel_id].pixel_status))
+
+        return mon
+
+    @staticmethod
+    def load_drs4_time_calibration_file(path):
```

Comment:
Code to create time calibration file will we leave in `cta-lstchain`?

Reply:
Yes, this repository will only be for the IO parts / application of the calibration to R1.